src/connectRoutes.js

```diff
@@ -115,7 +115,11 @@ export default (
   }
 
   const {
+    // Flow doesn't like defaulting to string or func here, although it does not
+    // complain if we _do not_ provide a default value
+    // $FlowIssue
```

Comment:
don't default there. default manually as part of a typeof function | string | *undefined* check (all in one place)